pos_event_sale/static/src/js/popups/EventTicketsPopup.js

```diff
@@ -54,6 +54,19 @@ odoo.define("pos_event_sale.EventTicketsPopup", function(require) {
             );
         },
 
+        getTicketFinalPrice: function(ticket) {
+            const pricelist = _.findWhere(self.posmodel.pricelists, {
+                name: this.pos.get_order().pricelist.name,
+            });
+            const product = this.pos.db.get_product_by_id(ticket.product_id[0]);
+            // When we sell events in pos, sales price is set at the ticket level and not in the product.
+            // We set product.lst_price with the price of the ticket. Like that, standard method get_price
+            // will be able to apply the discount on the top of ticket price.
+            // See get_price method: https://github.com/odoo/odoo/blob/44f7754af4fc82228f4f12d3f0b59a2cfe7ecab5/addons/point_of_sale/static/src/js/models.js#L1543
+            product.lst_price = ticket.price;
+            return product.get_price(pricelist, 1);
```

Comment:
Nice trick!

I'd be more comfortable if we leave things as we've found them:

```suggestion
            const product_lst_price = product.lst_price;
            product.lst_price = ticket.price;
            const price_reduce = product.get_price(pricelist, 1);
            product.lst_price = product_lst_price;
            return price_reduce;
```